src/dali/plugin/input/rest/kraken.py

```diff
@@ -145,46 +147,38 @@ def _process_gains(
         pass
 
     def _gather_api_data(self):
-        import pickle
-        import os
-
-        cache_exists = os.path.exists(self.__CACHE_FILE)
-        if self.use_cache and cache_exists:
-            with open(self.__CACHE_FILE, 'rb') as file:
-                something = pickle.load(file)
-                return something
+        loaded_cache = load_from_cache(self.__CACHE_FILE)
+        if self.use_cache and loaded_cache:
+            return loaded_cache
 
         # get initial trade history to get count
         index: int = 0
-        trade_history = self._client.private_post_tradeshistory(params={_OFFSET: index})
-        count: int = int(trade_history[_RESULT][_COUNT])
-        trade_history = dict()
+        count: int = int(self._client.private_post_tradeshistory(params={_OFFSET: index})[_RESULT][_COUNT])
+        trade_history: Dict[str, Dict[str, Union[str, int, None, List[str]]]] = {}
         while index < count:
             trade_history.update(self._process_trade_history(index))
             index += _TRADE_RECORD_LIMIT
 
-        ledger = self._client.private_post_ledgers(params={_OFFSET: index})
-        index = 0
-        count = int(ledger[_RESULT][_COUNT])
-        ledger = dict()
+        index: int = 0
+        count: int = int(self._client.private_post_ledgers(params={_OFFSET: index})[_RESULT][_COUNT])
+        ledger: Dict[str, Dict[str, Union[str, int, None, List[str]]]] = {}
         while index < count:
             ledger.update(self._process_ledger(index))
             index += _TRADE_RECORD_LIMIT
 
         result = (trade_history, ledger)
 
         if self.use_cache:
-            with open(self.__CACHE_FILE, 'wb') as file:
-                pickle.dump(result, file)
+            save_to_cache(self.__CACHE_FILE, result)
 
         return result
 
     def load(self) -> List[AbstractTransaction]:
         (trade_history, ledger) = self._gather_api_data()
 
-        result = {'in': [], 'out': [], 'intra': []}
+        result: Dict[str, List[AbstractTransaction]] = {'in': [], 'out': [], 'intra': []}
 
-        unhandled_types = dict()
+        unhandled_types: Dict[str, str] = {}
         for key, value in ledger.items():
             if value[_TYPE] == _WITHDRAWAL or value[_TYPE] == _DEPOSIT:
```

Comment:
Be sure to add a debug log with the native data: this is essential when debugging issues in the plugin. Here's an example from the Coinbase plugin: https://github.com/eprbell/dali-rp2/blob/main/src/dali/plugin/input/rest/coinbase.py#L452